specification/sql/resource-manager/Microsoft.Sql/preview/2022-02-01-preview/examples/ManagedInstanceDtcUpdateEnableDtc.json

```diff
@@ -0,0 +1,41 @@
+{
+  "parameters": {
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "testrg",
+    "managedInstanceName": "testinstance",
+    "dtcName": "current",
+    "api-version": "2022-02-01-preview",
+    "parameters": {
+      "properties": {
+        "dtcEnabled": true
+      }
+    }
```

Comment:
This looks like it is modelled as a patch , but I do not see any patch operation in your specification. Can you please clarify the intent here ?

Reply:
We use the PUT operation, but we only support the update scenario. We needed the PUT operation in order to enable polling via location and azureAsyncOperation headers.

Reply:
I've added a PATCH method also. Now the controller contains both patch and put requests.

Reply:
PUT needs to contain all the incoming required properties. PATCH can contain a subset of them. Please check any other definitions to see how PATCH is implemented for resrouces

Reply:
We don't have any required properties, that is why the examples are exactly the same. All the properties are entirely optional since that makes sense for the feature we are working on.

Reply:
I've forgotten to add that all the properties can be changed, except for "dtcHostNameDnsSuffix" which is declared as read only.

Reply:
Added new patch resource - ManagedInstanceDtcResourceProperties

Reply:
Removed the PATCH method and resources altogether since they offer the same functionality as PUT.